python/lib/scheduler/dmod/scheduler/resources/resource.py

```diff
@@ -288,28 +301,53 @@ def get_resource_enum_value(cls, enum_type: Union[Type[ResourceAvailability], Ty
                 return val
         return None
 
-    def __eq__(self, other):
+
+    def __eq__(self, other: object):
         if not isinstance(other, Resource):
             return super().__eq__(other)
         else:
             return self.resource_id == other.resource_id and self.hostname == other.hostname \
                    and self.availability == other.availability and self.state == other.state \
                    and self.cpu_count == other.cpu_count and self.memory == other.memory \
-                   and self.total_cpu_count == other.total_cpu_count and self.total_memory == other.total_memory
-
-    def __init__(self, resource_id: str, hostname: str, availability: Union[str, ResourceAvailability],
-                 state: Union[str, ResourceState], cpu_count: int, memory: int, total_cpu_count: Optional[int],
-                 total_memory: Optional[int]):
-        super().__init__(pool_id=resource_id, hostname=hostname, cpu_count=cpu_count, memory=memory)
-
-        self._availability = None
-        self.availability = availability
-
-        self._state = state
-        self.state = state
-
-        self._total_cpu_count = cpu_count if total_cpu_count is None else total_cpu_count
-        self._total_memory = memory if total_memory is None else total_memory
+                   and self.total_cpus == other.total_cpus and self.total_memory == other.total_memory
+
+    def __init__(
+        self,
+        resource_id: str = None,
+        hostname: str = None,
+        availability: Union[str, ResourceAvailability] = None,
+        state: Union[str, ResourceState] = None,
+        cpu_count: int = None,
+        memory: int = None,
+        total_cpu_count: Optional[int] = None,
+        total_memory: Optional[int] = None,
+        **data
+        ):
+        if data:
+            # NOTE: this can be removed alias field names _are_ case sensitive
+            potentially_aliased_fields = {
+                "availability": availability,
+                "hostname": hostname,
+                "state": state,
+                "total_memory": total_memory
+                }
+
+            for field_name, value in potentially_aliased_fields.items():
+                if value is not None:
+                    data[field_name] = value
+            super().__init__(**data)
+            return
+
+        super().__init__(
+            pool_id=resource_id,
+            hostname=hostname,
+            cpu_count=cpu_count,
+            memory=memory,
+            availability=availability,
+            state=state,
+            total_cpus=total_cpu_count,
+            total_memory=total_memory
+            )
```

Comment:
It would be more of a pain than anything. Ill add a comment to explain why there is conditional logic there. This is complex because in the previous code field names, `__init__` parameter names, and serialized key names (i.e. `to_dict`) all have minor naming differences. So, to retain backwards compatible and get all of that to line up correctly there is a little extra work involved. 

@robertbartel, this is related to a larger can of worms, but its worth considering simplifying our field names in places where there are discrepancies. That would largely remove logic like is present in `__init__`. That certainly is unrelated to the PR, but I think it is worth opening an issue to consider in the future. Thoughts?

Reply:
Given the `return` comes at the end of an `if` block, I'd expect the other `super` call to just be inside an `else`.  That it isn't makes me want to ask if there was a deliberate reason for not doing so.  @aaraney?

Reply:
@robertbartel, this is now a moot point. I removed the `__init__` method altogether. It was not being used elsewhere.